src/App.tsx

```diff
@@ -15,47 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = '',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
+interface SortingOptions {
+  sortMethod: SortType;
+  isReversed: boolean;
+}
+
 export const App: React.FC = () => {
+  const getGoodsToView = (
+    goods: string[],
+    { sortMethod, isReversed }: SortingOptions,
+  ) => {
+    let newGoods = [...goods];
+
+    switch (sortMethod) {
+      case SortType.Alphabetically:
+        newGoods = newGoods.sort((a, b) => a.localeCompare(b));
+        break;
+
+      case SortType.Length:
+        newGoods = newGoods.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        break;
+    }
+
+    if (isReversed) {
+      newGoods = newGoods.reverse();
+    }
+
+    return newGoods;
+  };
+
+  const [sortMethod, setSortMethod] = useState(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+  const GoodsToView: string[] = getGoodsToView(
```

Comment:
```suggestion
  const goodsToView: string[] = getGoodsToView(
```
we usually name variables in camelCase